File: backend/Adsetdesafio/Application/DTOs/CarsFilterDTO.cs
using Adsetdesafio.Shared.Utils.Enums;

namespace Adsetdesafio.Application.DTOs
{
    public class CarsFilterDTO
    {
        public int? AnoMin { get; set; }
        public int? AnoMax { get; set; }
        public long? Preco { get; set; } 
        public bool? SomenteComFotos { get; set; }
        public EnumCor? Cor { get; set; }
        public List<EnumOptionsCar>? Opcionais { get; set; }
        public Dictionary<string, EnumCategory>? PortaisECategorias { get; set; }
    }
}

File: frontend/adsetDesafioTecnico/src/app/shared/DTOs/CarsFilterDTO.ts
export interface CarsFilterDTO {
  placa?: string;
  marca?: string;
  modelo?: string;
  anoMin?: number;
  anoMax?: number;
  preco?: number;
  fotos?: string;
  opcionais?: string;
  cor?: string;
  page?: number;
  pageSize?: number;
  somenteComFotos:boolean;
}
